lib/puppet/network/http/base_pool.rb

```diff
@@ -4,7 +4,7 @@
 class Puppet::Network::HTTP::BasePool
   def start(site, verifier, http)
     Puppet.debug("Starting connection for #{site}")
-    if verifier
+    if site.use_ssl?
```

Comment:
just a clarifying question: If `site.use_ssl?` is false, `verifier` will be nil, right? So, this doesn't change behavior if we left it as it? I think this change is good because it's more clear what's happening, but I wanted to make sure I was on the same page with everything

Reply:
yeah the verifier should only be non-nil for ssl connections, so this shouldn't be a change in behavior. Currently `pool#with_connection` is called from https://github.com/puppetlabs/puppet/blob/613445a21ca8d5ceab84ca083d9690ffa1304f5e/lib/puppet/http/client.rb#L22-L29 and https://github.com/puppetlabs/puppet/blob/613445a21ca8d5ceab84ca083d9690ffa1304f5e/lib/puppet/network/http/connection.rb#L347. The latter only sets [`@verifier` to a non-nil value if configured to use ssl](https://github.com/puppetlabs/puppet/blob/613445a21ca8d5ceab84ca083d9690ffa1304f5e/lib/puppet/network/http/connection.rb#L60-L70).